Setting components in Component decorator causes Typescript 2.4 error

The following code worked in Typescript 2.3

```typescript
@Component({
  components: {
    Slideout,
    PanelComponent,
    VideoComponent
  }
})
export default class App extends Vue {
  @Prop()
  client: client;

  open() {
    console.log("Opened!");
  }
}
```

Using Typescript 2.4, I get the following error:
> ERROR in ./~/ts-loader!./src/App.ts
> (10,3): error TS2345: Argument of type '{ components: { 'Slideout': any; 'PanelComponent': "string" | "number" | "boolean" | "sym...' is not assignable to parameter of type 'VueClass'.
>  Object literal may only specify known properties, and 'components' does not exist in type 'VueClass'.

--- a/example/sfc.d.ts
+++ b/example/sfc.d.ts
@@ -1,4 +1,4 @@
 declare module "*.vue" {
   import Vue from 'vue'
-  export default typeof Vue
+  export default Vue
 }

